src/crystal/hasher.cr

```diff
@@ -45,12 +72,17 @@ struct Crystal::Hasher
   end
 
   def reference(value)
-    @result = @result * 31 + value.object_id.to_u64
-    self
+    value.object_id.to_u64.hash(self)
   end
 
   def string(value)
-    value.to_slice.hash(self)
+    @siphash.update(value)
+    self
+  end
+
+  def slice(value)
+    @siphash.update(value)
```

Comment:
It may be fine for `Slice(UInt8)` but probably :hankey: for anything else?

Reply:
@akzhan , it is debatable question.
`Slice[1, 2, 3].hash.should eq(Slice[1_u64, 2_u64, 3_u64].hash)` or should_not?